include/beman/scope/scope.hpp

```diff
@@ -113,7 +133,276 @@ make_unique_resource_checked(R&& r, const S& invalid, D&& d) noexcept(noexcept(
 //
 //        template <typename R, typename D>
 //        unique_resource(R, D) -> unique_resource<R, D>;
+//
+//==================================================================================================
+
+// Template argument `ScopeExitFunc` shall be
+//   - a function object type([function.objects]),
+//   - lvalue reference to function,
+//   - or lvalue reference to function object type.
+//
+//   If `ScopeExitFunc` is an object type, it shall meet the requirements of Cpp17Destructible(Table 30).
+//   Given an lvalue g of type remove_reference_t<EF>, the expression g() shall be well- formed.
+
+//==================================================================================================
+
+// --- Concepts ---
+template <typename F, typename R, typename... Args>
+concept invocable_return = std::invocable<F, Args...> && std::convertible_to<std::invoke_result_t<F, Args...>, R>;
+
+template <typename F>
+concept scope_exit_function =
+    invocable_return<F, void> && (std::is_nothrow_move_constructible_v<F> || std::is_copy_constructible_v<F>);
+
+template <typename T>
+concept scope_function_invoke_check = invocable_return<T, bool>;
+
+template <typename T>
+concept HasRelease = requires(T t) {
+    { t.release() } -> std::same_as<void>;
```

Comment:
I don't know if I had got this in the paper yet, but boost calls this something like ```disable``` which I think is a far superior name (apparently it came up in the boost review and is discussed in the section that compares boost with the TS).  This method doesn't 'release' anything, it just shuts off the end of scope behavior.  There's more changes in boost related to this -- like it can start disabled and only be enabled later which are options that deserve discussion.  But no matter what I don't like the release name -- I was confused as soon as I saw it...

Reply:
I agree that `release` can be a confusing name.
Another name for this is `cancel`.
`disable` is possibly a good comprimise.

The problem with this it has to be "hardcoded" at all.
If we inherit from the class containing this function, then the user can decide by themselves what name to give this function, in the context of there own code.
(Although this wil make a side-problem of how to pass the function name to the move function which needs to "release" the moved from object)